### PR TITLE
Fix PUP weaponskills and abilities

### LINKED ISSUE #126: Automaton Tank/Mage WS DMG is 0. Attachments are also still broken.
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/DerpyProjectGroup/topaz/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/DerpyProjectGroup/topaz/blob/base/CONTRIBUTING.md)
- [x] I have specified what branch this happens on: EVERYTHING

Sorry to bug you guys again haha! The last fixes did fix the ranged attacks and I apologize for not noticing this issue before.
- Automaton Tank frames WS deals 0 damage
- Automaton all Mage frame WS deals 0 damage
- Automaton Harlequin frames WS deals 0 damage
- Attachments are broken
   o Tried Several Attachments like Strobe/Shock Absorber/Flashbulb/etc with no luck

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

This PR is setting the stage for a continuing PUP refactor to bring things in line with retail.  There's some pieces included as well that start to lay the foundation for this effort.

Core:
* Add and wire up Automaton-specific ability functions for check and finish scripts
* Add caching for attachment script during item load
* Table automaton abilities (in DB) and add caching and storage for this data during map start

Lua:
* Fix automaton weaponskills not passing correct data to sub-functions
* Move automaton abilities to sub-directory in globals/abilities/pets
* Adjust ability scripts to reference new core functions for automaton-only checks and finishes

### CODE REVIEW

File: src/map/ai/controllers/automaton_controller.cpp [hunk 1]
```diff
@@ -1667,4 +1669,26 @@ namespace autoSpell
...
+        {
+            while (Sql_NextRow(SqlHandle) == SQL_SUCCESS)
+            {
+                uint16           id = (uint16)Sql_GetUIntData(SqlHandle, 0);
+                AutomatonAbility PAbility{ (uint8)Sql_GetUIntData(SqlHandle, 2), (uint16)Sql_GetUIntData(SqlHandle, 3) };
+
+                autoAbilityList[id] = std::move(PAbility);
+
+                auto filename = fmt::format("./scripts/globals/abilities/pets/automaton/{}.lua", Sql_GetData(SqlHandle, 1));
+                ShowDebug("Caching Automaton Ability: %s\n", filename);
```
Comment (zach2good) [inline src/map/ai/controllers/automaton_controller.cpp]: Can we get rid of this additional logging?

**Review by zach2good — APPROVED** [review]
Looks good to me

File: src/map/ai/controllers/automaton_controller.cpp (hunk 1 above)
Comment (claywar) [reply src/map/ai/controllers/automaton_controller.cpp]: Whoops! Absolutely

**Comment by claywar** [comment]
Going to rename `onAutomatonSkillCheck` to `onAutomatonAbilityCheck` for consistency later today.

**Pull request merged**